tests/test_metadata_serialization.py

```diff
@@ -29,9 +30,81 @@
 
 
+# pylint: disable=too-many-public-methods
 class TestSerialization(unittest.TestCase):
     """Test serialization for all classes in 'tuf/api/metadata.py'."""
 
+    invalid_signatures: utils.DataSet = {
+        "no signatures field": '{"signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}} \
+             }',
+        "missing keyid attribute in a signature": '{"signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}}, \
+            "signatures": [{ \
+                "sig": "abc" \
+              }] \
+            }',
+        "missing sig attribute in a signature": '{"signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}}, \
+            "signatures": [{ \
+                "keyid": "id" \
+              }] \
+            }',
+        "non unique duplicating signatures": '{"signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}}, \
+            "signatures": [ \
+                { \
+                    "keyid": "id", \
+                    "sig": "abc" \
+                }, \
+                { \
+                    "keyid": "id", \
+                    "sig": "abc" \
+                } \
+              ] \
+            }',
+        "unrecognized fields in signatures": '{"signed": \
+                { "_type": "timestamp", "spec_version": "1.0.0", "version": 1, "expires": "2030-01-01T00:00:00Z", \
+                "meta": {"snapshot.json": {"hashes": {"sha256" : "abc"}, "version": 1}}}, \
+            "signatures": [{ \
+                "keyid": "id", \
+                "sig": "abc", \
+                "foo": "bar" \
+              }] \
+            }',
+    }
+
+    @utils.run_sub_tests_with_dataset(invalid_signatures)
+    def test_invalid_signatures_serialization(self, test_data: str) -> None:
+        case_dict = json.loads(test_data)
+        # pylint: disable=no-member
+        if "unrecognized_fields" in self.case_name:
+            # When unrecognized fields exist the object creation will be
+            # successful, but the unrecognized fields won't be stored.
+            md = Metadata.from_dict(copy.deepcopy(case_dict))
+            self.assertNotEqual(case_dict, md.to_dict())
+        else:
+            with self.assertRaises((KeyError, ValueError)):
+                md = Metadata.from_dict(copy.deepcopy(case_dict))
+
+    valid_signatures: utils.DataSet = {
```

Comment:
this is also really valid_metadatas. I think valid_signatures (with actual signatures in it) might be more useful

I don't object to also add valid_metadatas but the possibility space explodes there... we  won't be able to test many meaningful combinations. but things like "multiple signatures", "no signatures" are still doable